lib/Maho.php

```diff
@@ -180,4 +180,25 @@ public static function errorReport(array $reportData = [], int $httpResponseCode
         header("HTTP/1.1 {$httpResponseCode} {$description}", true, $httpResponseCode);
         echo "<html><body><h1>There has been an error processing your request</h1>{$reportIdMessage}</body></html>";
     }
+
+    public static function getImageManager(array $customOptions = []): \Intervention\Image\ImageManager
+    {
+        $defaultOptions = [
+            'autoOrientation' => false,
+            'strip' => true,
+        ];
+        $options = [
+            ...$defaultOptions,
+            ...$customOptions,
+        ];
+
+        $driverClass = extension_loaded('vips') && \Composer\InstalledVersions::isInstalled('intervention/image-driver-vips')
```

Comment:
```suggestion
        $driverClass = extension_loaded('ffi') === true && ini_get('ffi.enable') === '1' && ini_get('zend.max_allowed_stack_size') === '-1' && \Composer\InstalledVersions::isInstalled('intervention/image-driver-vips')
```

I am needing all of these conditions for vips to be used. But also encountering another error...

Reply:
I am Ubuntu 22.04. This github answer helped me: https://github.com/libvips/php-vips/issues/251#issuecomment-2490828273

Without this change, if you put this debug line in lib/Maho.php, are you getting `Intervention\Image\Drivers\Vips\Driver`?

```php
        Mage::log("using $driverClass");
```

Reply:
I've to say I didn't test the vips variant.

but, could we use a function_exists like function_exists('vips_version') or something like that?

Reply:
I am pretty sure each of these conditions are needed to be true:

1. `extension_loaded('ffi')` -- should be built-in for most PHP distributions, but probably safer to check
2. `ini_get('ffi.enable')` -- default is `auto`, which will not work, must be `true`.
3. `ini_get('zend.max_allowed_stack_size')` -- must be -1 for PHP 8.3 or above

References: 
https://github.com/libvips/php-vips?tab=readme-ov-file#install

https://github.com/libvips/php-vips/issues/251#issuecomment-2494124994

Reply:
on my macos+brew it doesn't want to work :-(

Reply:
Actually, the `ffi.enable = preload` (the default) seems to be working, but the readme is specific that it's not yet supported, so I'm not sure if that's old.

Without `zend.max_allowed_stack_size` we do get the error `PHP Fatal error:  Uncaught Jcupitt\Vips\Exception: zend.max_allowed_stack_size not set to '-1'`, so potentially we don't need to check for that, it just depends if we want an error thrown or GD2 fallback.

Reply:
> on my macos+brew it doesn't want to work :-(

I added in my php.ini:

```
[PHP]
zend.max_allowed_stack_size = -1
ffi.enable = true
```

Reply:
So I have vips working, but my other problem is that after a few requests my server no longer responds until I `systemctl restart php8.3-fpm`. I had a similar problem before with my nginx + php-fpm stack, but I had resolved that so I'm not sure what's going on here. If we could make it work on your machine then I'd just chalk it up to something broken on my machine...

Is one of the conditions specifically failing for you? Or all these all `1`?

```php
Mage::log([
    extension_loaded('ffi') === true,
    ini_get('ffi.enable') === '1',
    ini_get('zend.max_allowed_stack_size') === '-1',
    \Composer\InstalledVersions::isInstalled('intervention/image-driver-vips'),
]);
```

Reply:
my fpm breaks completely on start but I can't see any proper log..

tomorrow I'll try to setup a containerized environment and see if I can make it work there :-)

Reply:
I am able to "fix" the problem on my end by setting `pm.max_requests = 1` in `/etc/php/8.3/fpm/pool.d/www.conf`, but this is a terrible idea for production.

Still trying to figure it out... I'd really like vips to work :)

Reply:
https://github.com/libvips/php-vips/issues/119 seemed to be a progress but it's not working